### ISSUE TITLE
depth option in richtext is not working

### ISSUE THREAD

**Issue by Stupidism** [payloadcms/payload#2937]
### Link to reproduction

Not for now

### To Reproduce

Try to fetch a relationship inside richtext with depth 2

### Describe the Bug

Before: 

<img width="1600" alt="image" src="https://github.com/payloadcms/payload/assets/5801015/16c4245f-fca4-417d-b204-5f725491cc40">

Now:

<img width="1665" alt="image" src="https://github.com/payloadcms/payload/assets/5801015/c14c671f-2b01-406a-9370-2c50c15d7743">

### Payload Version

1.10.1

**Comment by AlessioGr** [comment]
I'm able to reproduce the following:
- This ONLY happens with GraphQL. The local API works fine
- Despite having a depth of 2, 3, or anything higher, the result shows up as a depth of 1. I'm not able to populate it deeper than 1 with the depth

To reproduce:
1. Add a relation node to itself in a richtext field
2. Go to gql-playground: http://localhost:3000/api/graphql-playground
3. 
```
{
  Posts {
    docs {
      richtext(depth: 5)
    }
  }
}
```

**Comment by jmikrut** [comment]
Hey @Stupidism — thanks for the heads up here. This was a regression due to the `payloadAPI` now being truly respected throughout all operations. Basically, if the `payloadAPI` is set to GraphQL, it will now _**remain**_ at GraphQL, even throughout local populations. This conflicted with the way that we were setting `depth` for populations.

Fixed and coming out as 1.10.3 shortly!
